repos/kaleido/cc/scopes/Mermaid.h

```diff
@@ -0,0 +1,87 @@
+//
+// Created by coma007 on 4/4/24.
+//
+#include "Base.h"
+#include "base/bind.h"
+#include "base/command_line.h"
+#include "base/strings/string_util.h"
+#include "base/strings/stringprintf.h"
+#include "headless/public/devtools/domains/runtime.h"
+#include "../utils.h"
+#include <streambuf>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <fstream>
+
+#ifndef CHROMIUM_MERMAIDSCOPE_H
+#define CHROMIUM_MERMAIDSCOPE_H
+
+namespace kaleido {
+    namespace scopes {
+
+        class MermaidScope : public BaseScope {
+        public:
+            MermaidScope();
+
+            ~MermaidScope() override;
+
+            MermaidScope(const MermaidScope &v);
+
+            std::string ScopeName() override;
+
+            std::vector<std::unique_ptr<::headless::runtime::CallArgument>> BuildCallArguments() override;
+
+        };
+
+        MermaidScope::MermaidScope()  {
+
+            // Initialize mermaid object code
+            std::string mermaidInit = "mermaid.initialize( { startOnLoad: false, securityLevel: 'loose'} ); window.mermaid = mermaid;";
+
+            // Process mermaidjs
+            if (HasCommandLineSwitch("mermaidjs")) {
+                std::string mermaidjsArg = GetCommandLineSwitch("mermaidjs");
+
+                // Check if value is a URL
+                GURL mermaidjsUrl(mermaidjsArg);
+                if (mermaidjsUrl.is_valid()) {
+                    scriptTags.push_back(mermaidjsArg);
+                } else if (mermaidjsArg.find("import") != std::string::npos && mermaidjsArg.find("+esm") != std::string::npos) {
+                    // Add mermaid object initialization
+                    if (mermaidjsArg.find("mermaid.initialize") == std::string::npos) {
+                        mermaidjsArg += "; " + mermaidInit;
+                    }
+                    scriptTags.push_back(mermaidjsArg);
+                } else {
+                    // Check if this is a local file path
+                    if (std::ifstream(mermaidjsArg)) {
+                        localScriptFiles.emplace_back(mermaidjsArg);
+                    } else {
+                        errorMessage = base::StringPrintf("--mermaidjs argument is not a valid URL or file path: %s",
+                                                          mermaidjsArg.c_str());
+                        return;
+                    }
+                }
+            } else {
+                scriptTags.emplace_back("import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid/+esm'; " + mermaidInit);
+            }
+
```

Comment:
Not sure if I understand all the branching in this function, but it looks to me that it could be simplified. Let me know if I'm missing something, but I think that we can just check if the url has .mjs or esm+ extension to decide whether we should do localScriptFiles.emplace_back(mermaidjsArg) or we should instead emplace back "import mermaid from _mermaidjsArg ; window.mermaid = mermaid" (in both cases, whether we are checking the cdn file or local file path)

Reply:
Resolved [here](https://github.com/coma007/Kaleido/pull/1#discussion_r1553202694)

Fixed this, now only url is passed as the mermaidjs argument, and then if the module is esm, `import mermaid from '" + mermaidjsArg + "'"  + mermaidInit` is emplaced back to the `scriptTags`. 

I have talk about this in [another comment](https://github.com/coma007/Kaleido/pull/1#discussion_r1553190940):
```
std::string mermaidInit = "; mermaid.initialize( { startOnLoad: false, securityLevel: 'loose'} ); window.mermaid = mermaid;";
```